kahuna/public/js/search/index.js

```diff
@@ -129,7 +129,8 @@ search.config(['$stateProvider', '$urlMatcherFactoryProvider',
     });
 
     $stateProvider.state('search.results', {
-        url: 'search?{query:Query}&ids&since&nonFree&uploadedBy&until&orderBy',
+        url: 'search?{query:Query}&ids&since&nonFree&uploadedBy&until&orderBy' +
+             '&dateField&takenSince&takenUntil&modifiedSince&modifiedUntil',
```

Comment:
Good to know thanks. I think I'm going to leave as a String for now because that's how everything else works and I don't really want to refactor so much more unrelated stuff as part of this change..